src/Configuration/Dotenv.hs

```diff
@@ -54,53 +55,67 @@ loadFile
   :: MonadIO m
   => Config -- ^ Dotenv configuration
   -> m [(String, String)] -- ^ Environment variables loaded
-loadFile Config{..} = do
-  environment <- liftIO getEnvironment
-  readedVars <- concat `liftM` mapM parseFile configPath
-  neededVars <- concat `liftM` mapM parseFile configExamplePath
-  let coincidences = (environment `union` readedVars) `intersectEnvs` neededVars
+loadFile = runReaderT $ do
+  Config { .. } <- ask
+  environment <- (liftReaderT . liftIO) getEnvironment
+  readedVars <- liftReaderT $ fmap concat (mapM parseFile configPath)
+  neededVars <- liftReaderT $ fmap concat (mapM parseFile configExamplePath)
```

Comment:
Hey @FranzGB We can save the lifting by leaving the code as it was before and call `applySetting` with `runReaderT` at the end:

```suggestion
lloadFile config@Config{..} = do
  environment <- liftIO getEnvironment
  readedVars <- concat `liftM` mapM parseFile configPath
  neededVars <- concat `liftM` mapM parseFile configExamplePath
  ...
  runReaderT config (mapM applySetting vars)
```
Would that work?

Reply:
I think that could work I will change this in a bit 👍